pkgs/applications/networking/cluster/kops/default.nix

```
@@ -1,44 +1,54 @@
-{ stdenv, buildGoPackage, fetchFromGitHub, go-bindata }:
-
-buildGoPackage rec {
-  name = "kops-${version}";
-  version = "1.12.1";
+{ stdenv, lib, buildGoPackage, fetchFromGitHub, go-bindata }:
 
+let
   goPackagePath = "k8s.io/kops";
 
-  src = fetchFromGitHub {
-    rev = version;
-    owner = "kubernetes";
-    repo = "kops";
-    sha256 = "09rmgazdrmnh1lqaayzfbn0ld7mbj9whihs9ijv5gf6si9p0ml9y";
-  };
-
-  buildInputs = [go-bindata];
-  subPackages = ["cmd/kops"];
-
-  buildFlagsArray = ''
-    -ldflags=
-        -X k8s.io/kops.Version=${version}
-        -X k8s.io/kops.GitVersion=${version}
-  '';
-
-  preBuild = ''
-    (cd go/src/k8s.io/kops
-     go-bindata -o upup/models/bindata.go -pkg models -prefix upup/models/ upup/models/...)
-  '';
-
-  postInstall = ''
-    mkdir -p $bin/share/bash-completion/completions
-    mkdir -p $bin/share/zsh/site-functions
-    $bin/bin/kops completion bash > $bin/share/bash-completion/completions/kops
-    $bin/bin/kops completion zsh > $bin/share/zsh/site-functions/_kops
-  '';
-
-  meta = with stdenv.lib; {
-    description = "Easiest way to get a production Kubernetes up and running";
-    homepage = https://github.com/kubernetes/kops;
-    license = licenses.asl20;
-    maintainers = with maintainers; [offline zimbatm];
-    platforms = platforms.unix;
-  };
+  generic = { version, sha256, ...}@attrs:
+    let attrs' = builtins.removeAttrs attrs ["version" "sha256"] ; in
+      buildGoPackage ({
+        name = "kops-${version}";
+
+        inherit goPackagePath;
+        
+        src = fetchFromGitHub {
+          rev = version;
+          owner = "kubernetes";
+          repo = "kops";
+          inherit sha256;
+        };
+
+        buildInputs = [go-bindata];
+        subPackages = ["cmd/kops"];
+
+        buildFlagsArray = ''
+          -ldflags=
+              -X k8s.io/kops.Version=${version}
+              -X k8s.io/kops.GitVersion=${version}
+        '';
+
+        preBuild = ''
+          (cd go/src/k8s.io/kops
+           go-bindata -o upup/models/bindata.go -pkg models -prefix upup/models/ upup/models/...)
+        '';
+
+        postInstall = ''
+          mkdir -p $bin/share/bash-completion/completions
+          mkdir -p $bin/share/zsh/site-functions
+          $bin/bin/kops completion bash > $bin/share/bash-completion/completions/kops
+          $bin/bin/kops completion zsh > $bin/share/zsh/site-functions/_kops
+        '';
+
+        meta = with stdenv.lib; {
+          description = "Easiest way to get a production Kubernetes up and running";
+          homepage = https://github.com/kubernetes/kops;
+          license = licenses.asl20;
+          maintainers = with maintainers; [offline zimbatm];
+          platforms = platforms.unix;
+        };
+      } // attrs');
+in {
+  kops_1_12 = (generic {
```

the parenthesis is not needed here
```suggestion
  kops_1_12 = generic {
```